apps/versions/models.py

```diff
@@ -48,10 +53,17 @@ def get_query_set(self):
 
 
 def source_upload_path(instance, filename):
+    for ext in VALID_SOURCE_EXTENSIONS:
```

Comment:
Please add a comment above to explain that piece of code, for clarity. Something like:
`The source file should only have been saved if its extension was in VALID_SOURCE_EXTENSIONS.`